R/as_polars_df.R

```diff
@@ -74,7 +74,18 @@ as_polars_df <- function(x, ...) {
 #' @rdname as_polars_df
 #' @export
 as_polars_df.default <- function(x, ...) {
-  dtype <- infer_polars_dtype(x, ...)
+  dtype <- try_fetch(
+    infer_polars_dtype(x, ...),
+    error = function(cnd) {
+      abort(
+        sprintf(
+          "%s may not be converted to a polars Series, and hence to a polars DataFrame.",
```

Comment:
```suggestion
          "%s can't be converted to a polars Series, and hence to a polars DataFrame.",
```

Reply:
Strictly speaking, the failure of `infer_polars_dtype()` does not guarantee that `as_polars_series()` will fail.

Reply:
It doesn't? Do we have examples of this?

Reply:
For example, nanoarrow_array_stream.
I have added `as_polars_series` and `infer_polars_dtype` methods at the same time so `as_polars_df(<nanoarrow_array_stream>)` works, but if I register only `as_polars_series.nanoarrow_array_stream`, `infer_polars_dtype(<nanoarrow_array_stream>)` and `as_polars_df(<nanoarrow_array_tream>)` will fail.